presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -729,6 +736,65 @@ public void createTable(ConnectorSession session, ConnectorTableMetadata tableMe
                 new PartitionStatistics(basicStatistics, ImmutableMap.of()));
     }
 
+    @Override
+    public ConnectorTableHandle createTemporaryTable(ConnectorSession session, List<ColumnMetadata> columns, Optional<ConnectorPartitioningMetadata> partitioning)
+    {
+        String schemaName = getTemporaryTableSchema(session);
+        String tableName = "__presto_temporary_table_" + randomUUID().toString().replaceAll("-", "_");
+        HiveStorageFormat storageFormat = getTemporaryTableStorageFormat(session);
+        Optional<HiveBucketProperty> bucketProperty = partitioning.map(partitioning1 -> {
```

Comment:
>  I would just do a if

I prefer the functional style instead of `if`. When using functional style you don't have to call `Optional#get` all the time. So, i would prefer to keep the `partitioning.map`. 

> and inline with 749-757

That's a good idea. Will do.